Cleanup

Artturin opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Artturin committed:
tests.fetchurl: remove alias usage

Artturin committed:
fsfs: remove package which has never worked and project is dead

fsfs was added in https://github.com/NixOS/nixpkgs/commit/f00bdb6f150e2786bcb0186e95bf05b0d70ed44d
with a throw that states "it still does not build"

Artturin committed:
xxh: remove from pythonPackages set

error: xxh should use `buildPythonPackage` or `toPythonModule` if it is to be part of the Python pa
ckages set."}

Artturin committed:
tlspool: remove because its dependency gnutls-kdh has been removed

71276302bb3bd33972f1282ab1daff03dcf0584d
> gnutls-kdh: remove after being marked broken for over two years

Atemu commented on pkgs/tools/networking/xxh/default.nix:
```diff
@@ -1,19 +1,16 @@
 }:
 
-buildPythonApplication rec{
+python3.pkgs.buildPythonApplication rec{
```
Why not use `buildPythonPackage` or `toPythonModule` if that's what the error suggests? I don't think this is the correct way to package python applications such as this.

Artturin replied:
but this is a application, not a library

@pasqui23 opinions?

Atemu replied:
I've looked it up in the manual again and apparently this is indeed the intended way for packages that are pure applications (provide no python library).

Only for apps that also provide a library should you make them part of the `pythonPackages` set using `buildPythonPackage` and then `toPythonApplication` them in all-packages.

Artturin committed:
treewide: fix some Function called without required argument

by removing packages(if the dependency has been removed) or fixing the argument
